### PR TITLE
Added Radix Tree in data structures

### PR DESCRIPTION
**Pull request by acrulopez**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/trie/radix_tree.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                return incoming_node.is_leaf
+            # We have word remaining so we check the next node
+            else:
+                return incoming_node.find(remaining_word)
+
+    def delete(self, word: str) -> bool:
+        """Deletes a word from the tree if it exists
+
+        Args:
+            word (str): _description_
```
Comment (poyea) [inline data_structures/trie/radix_tree.py:128]: `_description_`

File: data_structures/trie/radix_tree.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        >>> RadixNode("myprefix").match("mystring")
+        ('my', 'prefix', 'string')
+        """
+        x = 0
+        for q, w in zip(self.prefix, word):
+            if q != w:
+                break
+
+            x += 1
+
```
Comment (cclauss) [inline data_structures/trie/radix_tree.py:36]:
```suggestion
        for x, (q, w) in enumerate(zip(self.prefix, word)):
            if q != w:
                break
```

**Review by cclauss — APPROVED** [review]
RSLGTM

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Pull request merged**